packages/catalog-process/src/services/catalogService.ts

```diff
@@ -420,15 +418,7 @@ export function catalogServiceBuilder(
       const eservice = await retrieveEService(eserviceId, readModelService);
       assertRequesterAllowed(eservice.data.producerId, authData.organizationId);
 
-      if (
-        !(
-          eservice.data.descriptors.length === 0 ||
-          (eservice.data.descriptors.length === 1 &&
-            eservice.data.descriptors[0].state === descriptorState.draft)
-        )
-      ) {
-        throw eServiceCannotBeUpdated(eserviceId);
-      }
```

Comment:
Refactor also in the `updateEService` endpoint, so the logic is cleaner. The only difference is what error is thrown (now it's `eserviceNotInDraftState`), but the status code is always 400.